src/detours.cpp

```diff
@@ -232,7 +232,43 @@ void FASTCALL Detour_UTIL_SayTextFilter(IRecipientFilter &filter, const char *pT
 		return UTIL_SayTextFilter(filter, pText, pPlayer, eMessageType);
 
 	char buf[256];
-	V_snprintf(buf, sizeof(buf), "%s %s", " \7CONSOLE:\4", pText + sizeof("Console:"));
+
+	if (!(g_bEnableZR && g_bTriggerTimer))
+	{
+		V_snprintf(buf, sizeof(buf), "%s %s", " \7CONSOLE:\4", pText + sizeof("Console:"));
+		UTIL_SayTextFilter(filter, buf, pPlayer, eMessageType);
+		return;
+	}
+
+	int iTriggerTimerLength = 0;
+	float fCurrentRoundClock = g_pGameRules->m_iRoundTime - (gpGlobals->curtime - g_pGameRules->m_fRoundStartTime.Get().m_Value);
+
+	for (int i = 0; i < strlen(pText); i++)
+	{
+		if (pText[i] >= '0' && pText[i] <= '9')
+			iTriggerTimerLength = (iTriggerTimerLength * 10) + (pText[i] - '0');
+
+		// Stop searching for more digits if a non-numerical character is detected after numerical characters
+		else if (iTriggerTimerLength > 0)	
+			break;
+	}
+
+	// Only display trigger time if the timer is greater than 4 seconds, and time expires within the round
+	if ((iTriggerTimerLength > 4) && (fCurrentRoundClock > iTriggerTimerLength))
+	{
+		int iTriggerTime = fCurrentRoundClock - iTriggerTimerLength;
+
+		// Round timer to nearest whole second
+		if ((int)(fCurrentRoundClock - 0.5f) == (int)fCurrentRoundClock)
+			iTriggerTime++;
+
+		int mins = iTriggerTime / 60;
+		int secs = iTriggerTime % 60;
+
+		V_snprintf(buf, sizeof(buf), "%s %s %s %2d:%02d", " \7CONSOLE:\4", pText + sizeof("Console:"), "\x10- @", mins, secs);
+	}
+	else
+		V_snprintf(buf, sizeof(buf), "%s %s", " \7CONSOLE:\4", pText + sizeof("Console:"));
```

Comment:
You should move all this to zr

Reply:
Vauff brought up a good point, this shouldn't be part of ZR even if only ZE servers will ever use it.

Reply:
Chat timer logic has been moved to its own function (currently in detours.cpp). The cvar has been renamed to `cs2f_trigger_timer_enable` and moved out of ZR.